aten/src/ATen/native/cuda/UnaryGeometricKernels.cu

```diff
@@ -21,15 +22,42 @@ void acos_kernel_cuda(TensorIteratorBase& iter) {
       });
 }
 
+const char asin_name[] = "asin";
 void asin_kernel_cuda(TensorIteratorBase& iter) {
-  AT_DISPATCH_FLOATING_AND_COMPLEX_TYPES_AND2(
+auto common_dtype = iter.common_dtype();
```

Comment:
minor nit: this is under-indented.

Reply:
Ah, right! Thanks, @kshitij12345! Have updated it :)